engine/evarutil.ml

```diff
@@ -193,7 +161,9 @@ let is_ground_env = memo is_ground_env
 
 exception NoHeadEvar
 
-let head_evar =
+let head_evar sigma c =
```

Comment:
This function should not take the sigma and be documented as an Unsafe (or better deprecated) API.
On the side one could add a function with a different name behaving in a (likely) more correct way, i.e. using kind_of_constr_upto.